pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java

```diff
@@ -58,21 +61,24 @@ public class LiteralTransformFunction implements TransformFunction {
   private String[] _stringResult;
   private byte[][] _bytesResult;
 
-  public LiteralTransformFunction(String literal) {
-    _literal = literal;
-    _dataType = inferLiteralDataType(literal);
-    if (_dataType.isNumeric()) {
-      _bigDecimalLiteral = new BigDecimal(_literal);
-    } else if (_dataType == DataType.BOOLEAN) {
-      _bigDecimalLiteral = PinotDataType.BOOLEAN.toBigDecimal(Boolean.valueOf(literal));
-    } else if (_dataType == DataType.TIMESTAMP) {
-      // inferLiteralDataType successfully interpreted the literal as TIMESTAMP. _bigDecimalLiteral is populated and
-      // assigned to _longLiteral.
-      _bigDecimalLiteral = PinotDataType.TIMESTAMP.toBigDecimal(Timestamp.valueOf(literal));
+  public LiteralTransformFunction(LiteralContext literalContext) {
+    Preconditions.checkNotNull(literalContext);
+    _literal = literalContext.getValue() == null ? "" : literalContext.getValue().toString();
+    if (literalContext.getType() == DataType.BOOLEAN) {
+      _bigDecimalLiteral = PinotDataType.BOOLEAN.toBigDecimal(literalContext.getValue());
+      _dataType = DataType.BOOLEAN;
     } else {
-      _bigDecimalLiteral = BigDecimal.ZERO;
+      _dataType = inferLiteralDataType(_literal);
```

Comment:
Do we still need to infer the type here given that `LiteralContext` now has the Pinot DataType

Reply:
Yes. Since this PR only adds support for boolean,  integers, big decimals, doubles etc still have a problem

Reply:
IMO we should not infer anymore. but for now we can keep this. and add a TODO:

I think this is a hacky way of "EXPRESSING" literal using String. keep in mind that we only can express number / floating number / varchar / bool in SQL syntax. but it doesn't prevent user from using something like
`CAST(1 AS DECIMAL) ` to construct a BIG_DECIMAL type literal context. 

we should add the rest in the future (but not along with the thrift conversion logic. as they will never be passed in as thrift, they will be optimized out by compileTimeLiteralFunctionInvoker

Reply:
added a TODO